plugin-server/src/main/ingestion-queues/buffer.ts

```diff
@@ -0,0 +1,70 @@
+import Piscina from '@posthog/piscina'
+import { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { Hub } from '../../types'
+import { runInstrumentedFunction } from '../utils'
+
+export function runBufferEventPipeline(hub: Hub, piscina: Piscina, event: PluginEvent) {
+    hub.lastActivity = new Date().valueOf()
+    hub.lastActivityType = 'runBufferEventPipeline'
+    return piscina.run({ task: 'runBufferEventPipeline', args: { event } })
+}
+
+export async function runBuffer(hub: Hub, piscina: Piscina): Promise<void> {
+    let eventRows: { id: number; event: PluginEvent }[] = []
+    await hub.db.postgresTransaction(async (client) => {
+        const eventsResult = await client.query(`
+            UPDATE posthog_eventbuffer SET locked=true WHERE id IN (
```

Comment:
Yeah so I wrote a few different approaches before settling on this - effectively moving races around.

I think I'll have this as a follow up + think about it tonight. Essentially as it stands this is _fine_ because at least we don't lose the event and can easily unlock these rows manually if need be.

But yeah, I'll think about solutions. 

Ideally this would all be wrapped in a transaction but we've been moving away from those. Could maybe do that with a timer? Anyway, just blabbering now.

Reply:
We can totally ship and iterate on this point later, though you might want to have the column be owner and a string rather than a boolean & filter over `owner = None`.